### PR TITLE
Fix: [iPad iOS 13] when having SSO code on clipboard, login page gets stuck in loop

### PR DESCRIPTION
**Pull request by billypchan**
### Issues

When a SSO code in copied to clipboard, it repeatedly appear even cancel is pressed.

### Causes

1. The `pollingTimer` keep creating alert even it is already shown
2. After cancel is pressed, it is still trying to show the alert when going to another screen.

### Solutions

1. Check if there is a exist SSO alert before presenting a new one
2. Stop polling after cancel is pressed, and set the flag `cancelSSOPressed` to true to prevent the alert shows again.

### CODE REVIEW

File: Wire-iOS/Sources/UserInterface/Company Login/CompanyLoginController.swift [hunk 1]
```diff
@@ -152,15 +163,27 @@ extension CompanyLoginController {
         error: UIAlertController.CompanyLoginError? = nil,
         ssoOnly: Bool = false) {
         
+        // Do not repeatly showing alert if exist
```
Comment (David-Henner) [inline Wire-iOS/Sources/UserInterface/Company Login/CompanyLoginController.swift]:
```suggestion
        // Do not repeatly show alert if exist
```

File: Wire-iOS/Sources/UserInterface/Company Login/CompanyLoginController.swift [hunk 2]
```diff
@@ -152,15 +163,27 @@ extension CompanyLoginController {
...
             ssoOnly: ssoOnly,
             error: error,
-            completion: { input in input.apply(inputHandler) }
+            completion: { [weak self] input, succeed in
+                input.apply(inputHandler)
+                self?.ssoAlert = nil
+                
+                // stop polling loop when cancel is pressed
+                if !succeed {
+                    self?.cancelSSOPressed = true
```
Comment (David-Henner) [inline Wire-iOS/Sources/UserInterface/Company Login/CompanyLoginController.swift]:
```suggestion
            completion: { [weak self] input, success in
                input.apply(inputHandler)
                self?.ssoAlert = nil
                
                // stop polling loop when cancel is pressed
                if !success {
                    self?.cancelSSOPressed = true
```

Comment (David-Henner) [inline Wire-iOS/Sources/UserInterface/Company Login/CompanyLoginController.swift]: Should we reset `cancelSSOPressed` somewhere?

File: Wire-iOS/Sources/UserInterface/Company Login/UIAlertController+CompanyLogin.swift [hunk 3]
```diff
@@ -86,8 +86,7 @@ extension UIAlertController {
         prefilledInput: String? = nil,
         ssoOnly: Bool = false,
         error: CompanyLoginError? = nil,
-        completion: @escaping (String?) -> Void
-        ) -> UIAlertController {
+        completion: @escaping (String?, Bool) -> Void) -> UIAlertController {
```
Comment (David-Henner) [inline Wire-iOS/Sources/UserInterface/Company Login/UIAlertController+CompanyLogin.swift]: could we, if possible, add parameter names for clarity ?

Comment (billypchan) [reply Wire-iOS/Sources/UserInterface/Company Login/UIAlertController+CompanyLogin.swift]: added labels.

File: Wire-iOS/Sources/UserInterface/Company Login/CompanyLoginController.swift (hunk 2 above)
Comment (billypchan) [reply Wire-iOS/Sources/UserInterface/Company Login/CompanyLoginController.swift]: `cancelSSOPressed` is reseted when the SSO alert is showing again (when the user pressed the enterprise log in button)

**Pull request merged**